src/widgets/widget-model.js

```diff
@@ -38,5 +43,20 @@ module.exports = cdb.core.Model.extend({
     this.dataviewModel.remove();
     this.trigger('destroy', this);
     this.stopListening();
+  },
+
+  setState: function (state) {
+    this.set(state);
+  },
+
+  getState: function () {
+    var state = {};
+    for (var key in this.defaultState) {
```

Comment:
`_.default` does the job of filling the `undefined` blanks, but it doesn't discard values that are equal to the default (I'm adopting the policy of not adding unchanged attributes to the state)